### PR TITLE
Adding messageFormat module for Utility functions using INTL 

### PR DESCRIPTION
**Pull request by RahulSubramaniam17**
[PD-191705](https://bazaarvoice.atlassian.net/browse/PD-191705)

Requirement

- [Design Doc ](https://bazaarvoice.atlassian.net/wiki/spaces/DEV/pages/78216462383/CFR+21+-+Display+Native+reviews+Design+Doc#Review-Location.1)for decision for the function 
- It was decided to add in bv-ui-core as it uses some browser API and does manipulation on top of that 
- Create a messageFormat module for all the utility functions that we use from INTL

Solution
Create messageFormat module 

[PD-191705]: https://bazaarvoice.atlassian.net/browse/PD-191705?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lib/messageFormat/index.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    //PolyFill for Intl.DisplayNames
+    if (typeof Intl === 'undefined' || typeof Intl.DisplayNames === 'undefined') {
+      // Load the polyfill if Intl or Intl.DisplayNames is not defined
+      const polyfillUrl = 'https://cdn.polyfill.io/v3/polyfill.min.js?features=Intl.DisplayNames';
+      const scriptEl = document.createElement('script');
+      scriptEl.src = polyfillUrl;
+      scriptEl.async = false;
+      document.head.appendChild(scriptEl);
+    }
+    // // Check if Intl.DisplayNames is supported
```
Comment (akshay-choulwar-94) [inline lib/messageFormat/index.js:28]: We can remove the commented code.

File: lib/messageFormat/index.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    // if (!Intl.DisplayNames) {
+    //   return '';
+    // }
+  
+    // Split the locale code into language and region components
+    const [language, region] = locale.split('_');
+  
+    // Get the display name for the region (i.e., the country name)
+    const regionNames = new Intl.DisplayNames([language], { type: 'region' });
```
Comment (akshay-choulwar-94) [inline lib/messageFormat/index.js:37]: What will happen if locale is invalid? Does regionNames will be undefined if yes then regionNames.of(region) I guess it will throw error.

Comment (akshay-choulwar-94) [inline lib/messageFormat/index.js:37]: What if the polyfill script failed to download new Intl will throw the error.

**Review by akshay-choulwar-94 — COMMENTED** [review]
Overall looks good to me. Added suggestions for edge cases. Also can we add unit tests for this.

File: lib/messageFormat/index.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+ * 
+ *  messageFormat can have functions which are used in multiple places
+ *
+ *  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl
+ */
+
+
+/**
+ * Generates the country name associated with the given locale
+ * @param {string} locale The locale code (e.g., 'en-US')
```
Comment (akshay-choulwar-94) [inline lib/messageFormat/index.js:14]: Do we supporting en_US and en-US format? Update the example given in the param document.

File: lib/messageFormat/index.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+ * @returns {string} The name of the country associated with the locale, or an empty string if the locale is not supported by Intl.DisplayNames
+ */
+export function generateCountryFromLocale(locale) {
+
+    //PolyFill for Intl.DisplayNames
+    if (typeof Intl === 'undefined' || typeof Intl.DisplayNames === 'undefined') {
```
Comment (akshay-choulwar-94) [inline lib/messageFormat/index.js:20]: We can use global.Intl here.
